### DISCUSSION TITLE
Improve MPPS forwarding capacities

### DISCUSSION THREAD

**Discussion by albertosaez** [dcm4che/dcm4chee-arc-light#3214] [Q&A] [answered]
We're trying to reuse the MPPS our central storage is receiving, by re-resending them to a dose managmente system. 

But we're facing two limitations with the current capacities of dcm4chee-arc :

- No capacity to filter which MPPS will be forwarded and which no. The solution to[ use different receiving aetitles](https://github.com/dcm4che/dcm4chee-arc-light/discussions/2943#discussioncomment-246884) is not possible as we will have to reconfigure all of the modalities. 

- No whay to change the "sending aetitltle" of the MPPS when forwarding, nor other customization like the ones are in the study exporter

Is there any plan to inlcude any of this capacities in the dcm4chee-arc ? 
Meanwhile, ... could you suggest any other tool impromve management of the  MPPS forwarding ?

Thanks, 

Alberto.

**Comment by gunterze** [comment, marked as answer]
Opened https://github.com/dcm4che/dcm4chee-arc-light/issues/3219

Masquerading of the sending AET is already configurable by [Network Extension](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/dcmNetworkAE.html) attribute:

Name | Type | Description (LDAP Attribute)
-- | -- | --
Masquerade Calling AE Title(s) | string | AE Title used for initiating network associations,  masquerading the actual AE Title for this Network AE - optional prefix  [&lt;Called AE Title&gt;] limits the masquerading to association to a  particular AE Title (dcmMasqueradeCallingAETitle)

Attribute Coercion is currently only supported on receive of N-CREATE/N-SET requests, which is also reflected in outgoing N-CREATE/N-SET requests. But you can't configure Attribute Coercions only applied on outgoing N-CREATE/N-SET requests dependent on MPPS attributes or the MPPS destination.
